mantidproject/mantid
# Indirect - Data Analysis Interfaces - Separate Mini-plot for Difference

Currently, the sample, fit and difference are all plotted in the same mini-plot. The difference should be plotted in a separate mini-plot below the existing one.

**To test:**
1. Navigate to Interfaces > Indirect > Data Analysis.
2. Follow the instructions outlined for the mentioned tabs:
    - **ConvFit**:
        1. Provide the supplied Sample (\*\_red) file with run number 26173 into the 'Sample' field.
        2. Provide the supplied Resolution (\*\_res) file into the 'Resolution' field.
        3. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        4. For each spectrum, ensure the Sample is plotted in the top mini-plot and the bottom mini-plot is empty.
        5. Select 'One Lorentzian' in the 'Fit Type' drop-down menu.
        6. Click 'Run'.
        7. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        8. For each spectrum, ensure the Sample and Fit are plotted in the top mini-plot and their difference is plotted in the bottom mini-plot.
        9. Provide the supplied Sample (\*\_red) file with run number 26176 into the 'Sample' field.
        10. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        11. For each spectrum, ensure the Sample is plotted in the top mini-plot and the bottom mini-plot is empty.
    - **IqtFit**:
        1. Provide the supplied Iqt (\*\_iqt) file with run number 26176 into the 'Input' field.
        2. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        3. For each spectrum, ensure the input is plotted in the top mini-plot and the bottom mini-plot is empty.
        4. Click 'Run'.
        5. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        6. For each spectrum, ensure the input and fit are plotted in the top mini-plot and their difference is plotted in the bottom mini-plot.
        7. Provide the supplied Iqt (\*\_iqt) file with run number 26173 into the 'Input' field.
        8. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        9. For each spectrum, ensure the Sample is plotted in the top mini-plot and the bottom mini-plot is empty.
    - **MSDFit**:
        1. Select any model in the 'Model' drop-down menu.
        2. Provide the supplied Sample (\*\_red) file with run number 26173 into the 'Input' field.
        3. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        4. For each spectrum, ensure the input is plotted in the top mini-plot and the bottom mini-plot is empty.
        5. Change the selected range with the blue dotted range selectors in the mini-plot (such that a non-zero range is selected).
        6. Click 'Run'.
        7. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        8. For each spectrum, ensure the input and fit are plotted in the top mini-plot and their difference is plotted in the bottom mini-plot.
        9. Provide the supplied Sample (\*\_red) file with run number 26176 into the 'Input' field.
        10. Navigate through each spectrum by changing the 'Plot Spectrum' spinner.
        11. For each spectrum, ensure the input is plotted in the top mini-plot and the bottom mini-plot is empty.
    - **JumpFit**:
        1. Provide the supplied Result (\*\_Result) file with run number 26173 into the 'Sample' field.
        2. Ensure the sample is plotted in the top mini-plot and the bottom mini-plot is empty.
        3. Click 'Run'.
        4. Ensure the fit and sample are plotted in the top mini-plot and the bottom mini-plot is empty.
        5. Provide the supplied Result (\*\_Result) file with run number 26176 into the 'Sample' field.
        6. Ensure the sample is plotted in the top mini-plot and the bottom mini-plot is empty.

## Test Files
**ConvFit**
[convfit-testfiles.zip](https://github.com/mantidproject/mantid/files/1390757/convfit-testfiles.zip)

**IqtFit**
[iqtfit-testfiles.zip](https://github.com/mantidproject/mantid/files/1390761/iqtfit-testfiles.zip)

**MSDFit**
[msdfit-testfiles.zip](https://github.com/mantidproject/mantid/files/1390763/msdfit-testfiles.zip)

**JumpFit**
[jumpfit-testfiles.zip](https://github.com/mantidproject/mantid/files/1390769/jumpfit-testfiles.zip)



Fixes #20241. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
